### PR TITLE
DOC: Clarify single-segment arrays in np reference

### PR DESCRIPTION
**Pull request by anirudh2290**
Clarified that single-segment and contiguous are synonymous. Please see: https://github.com/numpy/numpy/issues/15980

### CODE REVIEW

File: doc/source/reference/arrays.ndarray.rst [hunk 1]
```diff
@@ -133,6 +133,11 @@ Both the C and Fortran orders are :term:`contiguous`, *i.e.,*
 single-segment, memory layouts, in which every part of the
 memory block can be accessed by some combination of the indices.
 
+.. note::
+
+    `Contiguous arrays` and `single-segment arrays` are synonymous. They have been
+    used interchangeably in the numpy docs and have the same meaning.
```
Comment (rossbar) [inline doc/source/reference/arrays.ndarray.rst:139]:
```suggestion
    `Contiguous arrays` and `single-segment arrays` are synonymous
    and are used interchangeably throughout the documentation.
```

**Review by rossbar — APPROVED** [review]
LGTM - Seems to capture the discussion in #15980 and helps clear up some terminological uncertainty. I left a wording suggestion that I feel eliminates a bit of redundancy but it is fine as-is.

**Comment by anirudh2290** [comment]
Thank you @rossbar ! The suggestions look better, I have applied the suggestions.

**Review by WarrenWeckesser — APPROVED** [review]

**Comment by WarrenWeckesser** [comment]
Merged.  Thanks @anirudh2290 and @rossbar.

**Pull request merged**